datafusion/core/src/execution/context.rs

```diff
@@ -1130,7 +1131,7 @@ pub struct SessionConfig {
     /// Should DataFusion parquet reader using the predicate to prune data
     pub parquet_pruning: bool,
     /// Configuration options
-    pub config_options: ConfigOptions,
+    pub config_options: Arc<RwLock<ConfigOptions>>,
```

Comment:
Change `SessionConfig.config_options` from `ConfigOptions` to `Arc<RwLock<ConfigOptions>`
since
1. we need to share this with `CatalogWithInformationSchema`
2. allow to set config